### PR TITLE
Halloween race fixes

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
If you selected dullahan from the start screen, it would drop your head in the title area before warping you to station. This is not desirable.

Vampires will get their spells/not lose them on transform, and blood drain is slower

Fixes? Space bats not being able to fly in space

Fixes dullahan eyes action being named "generic action"

Fixes buckling issues with shapeshift. Fixes https://github.com/tgstation/tgstation/issues/32179

### CODE REVIEW

**Comment by ChangelingRain** [comment]
>This is not desirable.

oh?

**Comment by DoctoreMad** [comment]
>Changing vampires too

ATOOOOMIIIIIIIIIIC
Add it to the title at least

**Comment by Qustinnus** [comment]
150 hours

**Comment by Armhulen** [comment]
sorry you have negative credits gonna have to close this

**Comment by Armhulen** [comment]
unless you buff shadowpeople then i could probably pull some strings and get this in

**Comment by davethwave** [comment]
Isn't this a fix though?

**Comment by praisenarsie** [comment]
@Armhulen like you have any power in the codebase 😆

**Comment by SpaceManiac** [comment]
Dullahan swap vision spell is unnamed:
image

**Review by Cyberboss — DISMISSED** [review]
Fix the spell name

**Comment by chall8908** [comment]
Using the xenobio computers seems to be impossible as a dullahan.  The camera continues to follow your head instead of the computer.

File: code/modules/mob/living/carbon/human/species_types/vampire.dm [hunk 1]
```diff
@@ -127,4 +126,4 @@
 	shapeshift_type = /mob/living/simple_animal/hostile/retaliate/bat
```
Comment (AnturK) [inline code/modules/mob/living/carbon/human/species_types/vampire.dm]: All the lines below this one are unnecessary or plain wrong. Please remove.

File: code/modules/spells/spell_types/shapeshift.dm [hunk 2]
```diff
@@ -22,6 +22,11 @@
 		/mob/living/simple_animal/hostile/construct/armored)
 
 /obj/effect/proc_holder/spell/targeted/shapeshift/cast(list/targets,mob/user = usr)
+	if(src in user.mob_spell_list)
+		user.mob_spell_list.Remove(src)
+		user.mind.AddSpell(src)
```
Comment (AnturK) [inline code/modules/spells/spell_types/shapeshift.dm]: I think you meant to add it to the new body not new mind ?

Comment (KorPhaeron) [reply code/modules/spells/spell_types/shapeshift.dm]: If I add it to the mind then I can stop doing this silly process every time they transform

Comment (AnturK) [reply code/modules/spells/spell_types/shapeshift.dm]: But it will stay if they stop being a vampire ?

Comment (AnturK) [reply code/modules/spells/spell_types/shapeshift.dm]: Oh, nevermind, it's removed already.

**Pull request merged**